packages/modelserver-node/src/client/promise-utils.ts

```diff
@@ -0,0 +1,49 @@
+/********************************************************************************
+ * Copyright (c) 2022 STMicroelectronics.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0, or the MIT License which is
+ * available at https://opensource.org/licenses/MIT.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR MIT
+ *******************************************************************************/
+
+/**
+ * A simple protocol for a deferred value: an externally settlable promise.
+ */
+export interface Deferred<T> {
```

Comment:
How about `Future` in accordance to Java naming? the defer function could be named ComputableFuture ?
But I'm also fine with the deferred naming.

Reply:
Yeah. The problem is that the `Future` concept from Java actually maps on JS's `Promise`, so it would be confusing.

I'll go for `CompletablePromise`?